__ApplePie__ is a computer retail store that sells desktops and
laptops.


The company earns $\$$600 on each desktop computer it sells and $\$$900 on each laptop.


Due to COVID19 supply chain disruption, the supplier of ApplePie cannot ship more than 80 desktops and 75 laptops to ApplePie next month.


The employees at ApplePie must spend 2 hours installing software and checking each desktop computer they sell. They spend 3 hours to complete this process for laptop computers. They expect to have about 300 hours available for this purpose during the next month.


The store’s management is fairly certain that they can sell all the computers they order, but are unsure how many desktops and laptops they should order to __maximize profits__.

Let $x_1$ and $x_2$ be the number of desktop computers and laptop computers ApplePie order respectively.

\begin{align*}
    \text{Maximize   }&600x_1 + 900x_2&  \\
    \text{subject to   }&2x_1 + 3x_2\leq 300\\
    &x_1 \leq80\\
    &x_2 \leq 75\\
     & x_1,x_2 \geq 0
\end{align*}


We can represent the above problem with vectors and matrices.

\begin{align*}
    \text{Maximize   }&c^Tx&  \\
    \text{subject to   }&Ax\leq b\\
     & x \geq 0
\end{align*}

$$c = \begin{bmatrix}
600\\900
\end{bmatrix} \;\;\;  b = \begin{bmatrix}
300\\80\\75
\end{bmatrix}$$

$$A = \begin{bmatrix}
2&3\\1&0\\0&1
\end{bmatrix}$$

This helps us to encode the problem in R.

In [3]:
library(lpSolve)
c <- c(600, 900)
A <- matrix(c(2,3,1,0,0,1), nrow=3, byrow=T)
b <- c(300,80,75)
res <- lp("max", c, A, "<=", b)
res
res$solution

Success: the objective function is 90000 

[1] 37.5 75.0

According to the R output, the profit is maximised at 90000 dollars.
ApplePie will order 37.5 desktops and 75 laptops computers.